rfcs/table.md

```diff
@@ -0,0 +1,321 @@
+  name: 'Jane Smith',
+  age: 42,
+  address: 'London No. 1 Anywhere',
+}, {
+  key: '3',
+  name: 'Joe Black',
+  age: 32,
+  address: 'Sidney No. 1 Anywhere',
+}, {
+  key: '4',
+  name: 'Jane Red',
+  age: 32,
+  address: 'London No. 2 Anywhere',
+}];
+
+export default = () => <Table columns={columns} dataSource={data}/>,
+```
+
+#### Custom Table Rendering
+There may be a case where more complex layouts are needed,
+such as columns that with colSpan > 1 and/or rowspan > 1
+```
+export default() => {
+    <Table
+     render: (dataSource, columns) => {
+        return (<Tbody>
+          <Thead>
+            <TheadTr>
+              <Th>Item</Th>
+              <Th>Item</Th>
+              <Th>0.0000</Th>
+              <Th>Item</Th>
+            </TheadTr>
+          </Thead>
+          <Tbody>
+            <Tr>
+              <Th rowSpan={3}>Item</Th>
+              <Td>Item</Td>
+              <Td>0.0788</Td>
+              <Td>Item</Td>
+            </Tr>
+            <Tr>
+              <Td>Item</Td>
+              <Td>3.0210</Td>
+              <Td>Item</Td>
+            </Tr>
+            <Tr>
+              <Td>Item</Td>
+              <Td>4.0797</Td>
+              <Td>Item</Td>
+            </Tr>
+            <Tr>
+              <Th rowSpan={2}>Item</Th>
+              <Td>Item</Td>
+              <Td>0.0640</Td>
+              <Td>Item</Td>
+            </Tr>
+            <Tr>
+              <Td>Item</Td>
+              <Td isNumerical={true}>1.0117</Td>
+              <Td>Item</Td>
+            </Tr>
+          </Tbody>
+       );
+     }
+    />
+}
+```
+
+### Advanced Usages
+
+#### Remote Data
+Paginate data from a remote backend.
+
+```javascript
+import * as React from 'react';
+import {Table} from 'baseui/table';
+import {Pagination} from 'baseui/pagination';
+
+const columns = {
+    {
+        title: 'Employee ID'
+        dataIndex: 'id'
+    },
+    {
+        title: 'First'
+        dataIndex: 'firstName'
+    },
+    {
+        title: 'Last'
+        dataIndex: 'lastName'
+    },
+    {
+        title: 'Full Name'
+        render: (rowData) => {
+            const {firstName, lastName} = rowData;
+            return (
+                <span>
+                    `${firstName} ${lastName}`
+                </span>
+            );
+        }
+    }
+};
+
+class TableWithPagination extends React.Component {
+    state = {
+        currentPage: 1,
+        pageSize: 10,
+        data: [],
+        numPages: 0
+    };
+    fetch = (pageNumber, pageSize) => {
+        // logic to fetch data based on page number
+        // `currentPage` and `data` and `numPages` will be updated here.
+    }
+    componentDidMount = () => {
+        const {currentPage, pageSize} = this.state;
+        this.fetch(currentPage, pageSize);
+    }
+    onPageChange = (nextPage, prevPage) => {
+        const {pageSize} = this.state;
+        this.fetch(nextPage, pageSize);
+    }
+    render = () => {
+         return (
+             <Table
```

Comment:
Seems like there should be some loading state in the table with remote data

Reply:
okay, I'll add an `isLoading` prop to the table.